packages/fxa-shared/sentry/index.ts

```diff
@@ -1,3 +1,68 @@
 export * from './tag';
 export * from './config-builder';
 export * from './models/SentryConfigOpts';
+import { Hub, Severity } from '@sentry/types';
+
+/** Minimal type containing validation error fields that are used by routines. */
+export type ValidationError = {
+  details: {
+    message: string;
+    path: string[];
+    type: string;
+  }[];
+};
+
+/**
+ * Format a Stripe product/plan metadata validation error message for
+ * Sentry to include as much detail as possible about what metadata
+ * failed validation and in what way.
+ *
+ * @param {string} planId
+ * @param {string | import('@hapi/joi').ValidationError} error
+ */
+export function formatMetadataValidationErrorMessage(
+  planId: string,
+  error: ValidationError
+) {
+  let msg = `${planId} metadata invalid:`;
+  if (typeof error === 'string') {
+    msg = `${msg} ${error}`;
+  } else {
+    msg = `${msg}${error.details
+      .map(({ message }) => ` ${message};`)
+      .join('')}`;
+  }
+  return msg;
+}
+
+/**
+ * Report a validation error to Sentry with validation details.
+ *
+ * @param {Pick<Hub, 'withScope' | 'captureMessage'>} sentry - Current sentry instance. Note, that this subtype is being
+ *    used instead of directly accessing the sentry instance inorder to be context agnostic.
+ * @param {*} message
+ * @param {string | import('@hapi/joi').ValidationError} error
+ */
+export function reportValidationError(
+  sentry: Pick<Hub, 'withScope' | 'captureMessage'>,
```

Comment:
Type might seem a bit strange here, but see comment in doc about being context agnostic. Essentially, we don't want to make assumptions about the type of sentry instance (e.g. that is a '@sentry/node' instance).

Reply:
We're only using Sentry as a module global, are we expecting to pass around confined instances of it now?

Reply:
See other comment about making sentry context agnostic. Maybe we can side bar on this? I agree this one off approach is not ideal.

Reply:
I was able to work around the context issue by moving these functions out of index.ts.